doc/source/packaging/build-systems.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_build_system:
+
+############
+Build System
+############
+
+The build system is a fundamental tool for packaging Python
+libraries. It generates distribution files that can be shared with
+users and developers. 
+
+
+Artifacts
+=========
+
+The build-system allows maintainers to generate artifacts for their Python
+libraries. Here, `artifacts` refers to both wheel and source files:
+
+- ``Wheel files`` have the ``*.whl`` file extension.
+- ``Source files`` have the ``*.tar.gz`` or ``*.zip`` extension.
+
+These are the files to upload to `PyPI`_ when releasing a new version of a
+PyAnsys project.
+
+.. warning::
+
+   Not all files are included by default in the source distribution. A
+   `MANIFEST.in`_ is required at the root of the project to specify additional
+   files.
+
+
+The interaction between the maintainer and the build-system is performed using a
+build-system tool. This tool provides a frontend and a backend. The maintainers
+trigger the frontend, which then calls the backend. The backend then reads the
+project directory and generates the artifacts, as :numref:`build system diag`:
+
+.. include:: diag/build_system_diag.rst
+
+
+PEP 517 and PEP 518
+===================
+
+For a long time, the ``setup.py`` file was the only way of specifying the
+project structure, metadata, and installation workflow that `pip`_ was to follow.
+However, having to execute a Python file when installing a Python package
+introduced the following problems:
+
+- It was not possible to know which dependencies required the ``setup.py`` file
+  to be properly executed.
+
+- The default Python package installer, `pip`_, expected `setuptools`_ to be the
+  default build system tool, excluding others like `flit`_ and `poetry`_.
+
+These problems led to the acceptance of `PEP 517`_ and `PEP 518`_.
+
+
+PEP 517
+-------
+
+`PEP 517`_ allows Python developers to specify the build-backend tool for
+generating artifacts. As seen in the :numref:`build system diag` figure, these
+are the most popular backends:
+
+- `Setuptools`_ , while very popular, lacks build time dependency declaration
+  and is difficult to extend.
```

Comment:
```suggestion
- `Setuptools`_ , while very popular, lacks the ability to declare build time dependencies
  and is difficult to extend.
```

Reply:
@jorgepiloto If "build time dependency declaration" is a common developer term, you can ignore this comment. The general guideline is to limit the adjectives that modify a noun to two to keep it more readable, comprehensible, and, translatable. However, exceptions exist for terms that are commonly used and understood.